mindsdb/interfaces/skills/skill_tool.py

```diff
@@ -51,32 +52,42 @@ def _make_text_to_sql_tools(self, skill: db.Skills) -> dict:
         '''
            Uses SQLAgent to execute tool
         '''
-
+        # To prevent dependency on Langchain unless an actual tool uses it.
+        try:
+            from mindsdb.integrations.handlers.langchain_handler.mindsdb_database_agent import MindsDBSQL
+            from langchain_community.agent_toolkits.sql.toolkit import SQLDatabaseToolkit
+            from langchain_community.tools.sql_database.tool import QuerySQLDataBaseTool
+            from langchain_community.chat_models import ChatOpenAI
+        except ImportError:
+            raise ImportError('To use the text-to-SQL skill, please install langchain with `pip install mindsdb[langchain]`')
         database = skill.params['database']
         tables = skill.params['tables']
-
-        sql_agent = SQLAgent(
-            self.get_command_executor(),
-            database,
-            include_tables=tables
-        )
-
-        description = (
-            "Use the conversation context to decide which table to query. "
-            "Input to this tool is a detailed and correct SQL query, output is a result from the database. "
-            "If the query is not correct, an error message will be returned. "
-            "If an error is returned, rewrite the query, check the query, and try again. "
-            f"These are the available tables: {','.join(tables)}\n"
-        )
-        for table in tables:
-            description += f'Table name: "{table}", columns {sql_agent.get_table_columns(table)}\n'
-
-        return dict(
-            name='sql_db_query',
-            func=sql_agent.query_safe,
-            description=description,
-            type=skill.type
+        tables_to_include = [f'{database}.{table}' for table in tables]
+        db = MindsDBSQL(
+            engine=self.get_command_executor(),
+            metadata=self.get_command_executor().session.integration_controller,
+            include_tables=tables_to_include
         )
+        # Users probably don't need to configure this for now.
+        llm = ChatOpenAI(model=_DEFAULT_SQL_LLM_MODEL, temperature=0)
```

Comment:
Are sure this won't cause issues if user hasn't got openai env variabel set? see https://github.com/mindsdb/mindsdb/pull/9080/files

Reply:
And what if a user want to use not openai model?